### ISSUE TITLE
task: support re-run over time range

### ISSUE THREAD

**Issue by mark-rushakoff** [influxdata/platform#613]
Task backend issue to support #611.

The current scheduler assumes time is always moving forward. We need to support re-runs of a requested time range. Supporting this case should simplify retries of a single run.

**Comment by mark-rushakoff** [comment]
Here's roughly how I see this being implemented:

- New task/backend.Store method like `ManuallyRunTimeRange(taskID platform.ID, start, stop int64) error`, that creates a new [StoreTaskMetaManualRun](https://github.com/influxdata/platform/blob/be7a3e56dee504b5eaf61ce25e899aa4dc8578ba/task/backend/meta.pb.go#L167-L178) entry in the appropriate StoreTaskMeta. The field is write only for this commit or PR.
- Update [(*StoreTaskMeta).CreateNextRun](https://github.com/influxdata/platform/blob/be7a3e56dee504b5eaf61ce25e899aa4dc8578ba/task/backend/meta.go#L63-L64): a "naturally due" run takes precedence, then if there is a manual run queued, add it to stm.CurrentlyRunning, and set [RangeStart, RangeEnd, and RequestedAt fields](https://github.com/influxdata/platform/blob/be7a3e56dee504b5eaf61ce25e899aa4dc8578ba/task/backend/meta.pb.go#L112-L117) so that we can tell it was a run from a manual request, not a naturally scheduled run. The `StoreTaskMetaManualRun.LatestCompleted` field needs to move forward to indicate that we've progressed in the queue. If this was the last run of the queue, remove the `StoreTaskMetaManualRun` entry.

Then we're waiting on https://github.com/influxdata/platform/issues/611 to connect this to the rest of the API.

**Comment by mark-rushakoff** [comment]
Closed in #698. 

#611 is still outstanding.
